NLU.DevOps tools System.Net.Http.HttpRequestException: Resource temporarily unavailable

I've received below error in my pipeline in "Test Luis Model" step. When I try to run this command in my local environment, it works without a problem.

I'm using westeurope endpoint.

![image](https://user-images.githubusercontent.com/118657/81608590-35edfa00-93df-11ea-979f-6435e5e32c39.png)


`info: NLU.DevOps.CommandLine.Test.TestCommand[0]
      Running tests against NLU model...

   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.Rest.RetryDelegatingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime.PredictionOperations.GetVersionPredictionWithHttpMessagesAsync(Guid appId, String versionId, PredictionRequest predictionRequest, Nullable`1 verbose, Nullable`1 showAllIntents, Nullable`1 log, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime.PredictionOperationsExtensions.GetVersionPredictionAsync(IPredictionOperations operations, Guid appId, String versionId, PredictionRequest predictionRequest, Nullable`1 verbose, Nullable`1 showAllIntents, Nullable`1 log, CancellationToken cancellationToken)
   at NLU.DevOps.Luis.LuisTestClient.QueryAsync(PredictionRequest predictionRequest, CancellationToken cancellationToken) in D:\a\1\s\src\NLU.DevOps.LuisV3\LuisTestClient.cs:line 51
   at NLU.DevOps.Luis.LuisNLUTestClient.TestAsync(JToken query, CancellationToken cancellationToken) in D:\a\1\s\src\NLU.DevOps.LuisV3\LuisNLUTestClient.cs:line 53
   at NLU.DevOps.CommandLine.EnumerableExtensions.<>c__DisplayClass0_0`2.<<SelectAsync>g__selectWithIndexAsync|1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at NLU.DevOps.CommandLine.EnumerableExtensions.SelectAsync[T,TResult](IEnumerable`1 items, Func`2 selector, Int32 degreeOfParallelism) in D:\a\1\s\src\NLU.DevOps.CommandLine\EnumerableExtensions.cs:line 36
   at NLU.DevOps.CommandLine.Test.TestCommand.RunAsync() in D:\a\1\s\src\NLU.DevOps.CommandLine\Test\TestCommand.cs:line 51
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at NLU.DevOps.CommandLine.Test.TestCommand.Main() in D:\a\1\s\src\NLU.DevOps.CommandLine\Test\TestCommand.cs:line 29
   at NLU.DevOps.CommandLine.Program.Run(ICommand command) in D:\a\1\s\src\NLU.DevOps.CommandLine\Program.cs:line 36
   at NLU.DevOps.CommandLine.Program.Main(String[] args) in D:\a\1\s\src\NLU.DevOps.CommandLine\Program.cs:line 18
##[error]Process completed with exit code 134.`

--- a/.github/workflows/luis_ci.yaml
+++ b/.github/workflows/luis_ci.yaml
@@ -58,13 +58,13 @@ jobs:
 
     - name: Get LUIS authoring key
       run: |
-         keya=$(az cognitiveservices account keys list --name ${{ secrets.AZURE_LUIS_AUTHORING_RESOURCE_NAME }} --resource-group $AzureResourceGroup --query "key1" | xargs)
+         keya=$(az cognitiveservices account keys list --name ${{ secrets.AZURE_LUIS_AUTHORING_RESOURCE_NAME }} --resource-group ${{ secrets.AZURE_RESOURCE_GROUP }} --query "key1" | xargs)
          echo "::set-env name=LUISAuthoringKey::$keya"
          echo "::add-mask::$keya"
 
     - name: Get LUIS prediction key
       run: |
-         keyp=$(az cognitiveservices account keys list --name ${{ secrets.AZURE_LUIS_PREDICTION_RESOURCE_NAME }} --resource-group $AzureResourceGroup --query "key1" | xargs)
+         keyp=$(az cognitiveservices account keys list --name ${{ secrets.AZURE_LUIS_PREDICTION_RESOURCE_NAME }} --resource-group ${{ secrets.AZURE_RESOURCE_GROUP }} --query "key1" | xargs)
          echo "::set-env name=LUISPredictionKey::$keyp"
          echo "::add-mask::$keyp"
 
@@ -110,8 +110,23 @@ jobs:
       run: bf luis:version:import --appId $LUISAppId --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey  --in model.json
 
     - name: Train luis
-      shell: bash
-      run: bf luis:train:run --appId $LUISAppId --versionId $LUISAppVersion --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey  --wait
+      run: bf luis:train:run --appId $LUISAppId --versionId $LUISAppVersion --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey
+
+    - name: Wait for train
+      run: |
+        while true
+        do 
+          status=`bf luis:version:list --appId $LUISAppId --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey |  jq '.[] | select(.version == '\"$LUISAppVersion\"') | .trainingStatus' | xargs`
+          if [ "$status" == "Trained" ]
+          then
+            break
+          elif [ "$status" == "Failed" ]
+          then
+            exit 1
+          fi
+          echo 'sleep 10'
+          sleep 10
+        done 
 
     - name: Publish luis
       run: |

--- a/.github/workflows/luis_pr.yaml
+++ b/.github/workflows/luis_pr.yaml
@@ -69,7 +69,23 @@ jobs:
         jq '.id' | xargs -I {} echo "::set-env name=LUISAppId::{}"
         
     - name: Train luis
-      run: bf luis:train:run --appId $LUISAppId --versionId 0.1 --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey --wait
+      run: bf luis:train:run --appId $LUISAppId --versionId 0.1 --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey
+
+    - name: Wait for train
+      run: |
+        while true
+        do 
+          status=`bf luis:version:list --appId $LUISAppId --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey |  jq '.[] | select(.version == "0.1") | .trainingStatus' | xargs`
+          if [ "$status" == "Trained" ]
+          then
+            break
+          elif [ "$status" == "Failed" ]
+          then
+            exit 1
+          fi
+          echo 'sleep 10'
+          sleep 10
+        done 
 
     - name: Publish luis
       run: bf luis:application:publish --appId $LUISAppId --versionId 0.1 --endpoint $LUISAuthoringEndpoint --subscriptionKey $LUISAuthoringKey

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -19,6 +19,7 @@
         "dataset",
         "devops",
         "docx",
+        "elif",
         "eval",
         "exportlu",
         "githubsecretsall",

